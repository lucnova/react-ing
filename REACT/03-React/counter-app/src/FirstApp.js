import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ title, description, aBoolean }) => {
	const imgSrc = 'https://pbs.twimg.com/media/EqNM6G3XYAAUe14.jpg';
	//const imgSrc = 'https://instagram.fgdl1-3.fna.fbcdn.net/v/t51.2885-15/e35/146082774_1379663119035761_1060428450675773640_n.jpg?_nc_ht=instagram.fgdl1-3.fna.fbcdn.net&_nc_cat=105&_nc_ohc=VCD0ubqCuIMAX9wEbpz&tp=1&oh=d4a070cce29f651cea031e585bc7525e&oe=604673A6';

	return (
		<>
			<h1>{title}</h1>
			<p>{description}</p>
			<img src={imgSrc} alt={description}></img>
		</>
	);
};

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	description: PropTypes.string,
	aBoolean: PropTypes.bool,
};

/* VALORES POR DEFECTO */
FirstApp.defaultProps = {
	aBoolean: true,
};

export default FirstApp;
