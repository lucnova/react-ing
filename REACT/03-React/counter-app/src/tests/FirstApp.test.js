/**
 * Probando Componentes con JEST
 */
//import { render } from '@testing-library/react'; // - Pruebas Nativas -
import '@testing-library/jest-dom'; // Para probar DOM en JEST
import { shallow } from 'enzyme';

import FirstApp from '../FirstApp'; // Componente a probar

describe('Testing <FirstApp /> Component', () => {
	// - ENZYME -
	test('SHOULD show <FirstApp /> correctly', () => {
        const expectedTitle = 'Yo Opino';
		const wrapper = shallow( <FirstApp title={expectedTitle}/> );

        expect(wrapper).toMatchSnapshot();
	});

	// - Pruebas Nativas -
	/*
	test('MUST be string: "Yo Opino"', () => {
		const expected = 'Yo Opino';

		// - Sin desestructurar -
		//const wrapper = render(<FirstApp title={'Yo Opino'} description={'Que Opinar es Necesario'} />); // Producto renderizado de lo que se quiere evaluar
		//wrapper.getByText();

		// - Destructurando -
		const { getByText } = render(<FirstApp title={expected} description={'Que Opinar es Necesario'} />); // Producto renderizado de lo que se quiere evaluar

		// - wrapper tiene varias funciones para testear.

		expect(getByText(expected)).toBeInTheDocument();

        // SE USAR√Å ENZYME
	});
    */
});
