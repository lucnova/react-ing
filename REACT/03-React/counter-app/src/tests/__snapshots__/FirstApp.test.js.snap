// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <FirstApp /> Component SHOULD show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Yo Opino
  </h1>
  <p />
  <img
    src="https://pbs.twimg.com/media/EqNM6G3XYAAUe14.jpg"
  />
</Fragment>
`;
